src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporter.cs

```diff
@@ -548,64 +548,68 @@ private static void FormatStandardAndErrorOutput(ITerminal terminal, string? sta
     private void AppendAssemblyLinkTargetFrameworkAndArchitecture(ITerminal terminal)
     {
         terminal.AppendLink(_assembly, lineNumber: null);
-        if (_targetFramework != null || _architecture != null)
+        if (_targetFramework == null && _architecture == null)
         {
-            terminal.Append(" (");
-            if (_targetFramework != null)
-            {
-                terminal.Append(_targetFramework);
-                terminal.Append('|');
-            }
+            return;
+        }
 
-            if (_architecture != null)
-            {
-                terminal.Append(_architecture);
-            }
+        terminal.Append(" (");
+        if (_targetFramework != null)
+        {
+            terminal.Append(_targetFramework);
+            terminal.Append('|');
```

Comment:
Unrelated to your change, but maybe we should only append `|` if both TFM and arch are non-null. Otherwise, we can end up with `(net8.0|)`

Reply:
Let me do a follow-up.